### PR TITLE
feat: add `math/base/special/sqrtpif`

### PR DESCRIPTION
**Pull request by gururaj1512**
Resolves a part of #649 .

## Description

> What is the purpose of this pull request?

Adds package `math/base/special/sqrtpif`

This pull request:

-   Progresses: #649 

## Related Issues

> Does this pull request have any related issues?

- Test cases are failing due to values unaccurate more than tolerance.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by gururaj1512** [comment]
@gunjjoshi, The test cases are failing for this package. Can you review `main.js` and point out if something is wrong.

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var sqrtf = require( '@stdlib/math/base/special/sqrtf' );
+var PI = require( '@stdlib/constants/float32/pi' );
+
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js:26]:
```suggestion
var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );

```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* // returns 0.0
+*
+* v = sqrtpif( -4.0 );
+* // returns NaN
+*
+* v = sqrtpif( NaN );
+* // returns NaN
+*/
+function sqrtpif( x ) {
+	return sqrtf( x * PI );
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js]:
```suggestion
	return sqrtf( float64ToFloat32( float64ToFloat32( x ) * PI ) );
```

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js]: You should use `float64ToFloat32` to wrap the input values and arithmetic operations, when dealing with single-precision calculations.

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js]: This is similar to what we follow in `math/base/special/sqrtf`: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/sqrtf/lib/main.js

Comment (gururaj1512) [reply lib/node_modules/@stdlib/math/base/special/sqrtpif/lib/main.js]: > You should use `float64ToFloat32` to wrap the input values and arithmetic operations, when dealing with single-precision calculations.

I have tried this implementation; the test cases are failing.
image
image

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/test/fixtures/julia/runner.jl [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+``` julia
+julia> x = range( -1000, stop = 1000, length = 2001 );
+julia> gen( x, \"data.json\" );
+```
+"""
+function gen( x, name )
+	y = Array{Float32}( undef, length(x) );
+	for i in eachindex(x)
+		y[i] = sqrt( Float32( x[i] * pi ) );
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/test/fixtures/julia/runner.jl]: @gururaj1512 I tried running these tests locally. In order to fix your failing tests, you'll need to change this to:
```suggestion
		y[i] = sqrt( Float32( Float32( x[i] ) * pi ) );
```
We first cast `x[ i ]` to `Float32`, and then cast its product with `pi` to `Float32` again.
After changing this, you'll need to regenerate the fixtures, and all tests should pass after this.

Comment (gururaj1512) [reply lib/node_modules/@stdlib/math/base/special/sqrtpif/test/fixtures/julia/runner.jl]: Thanks for clarifying. This is working well.

**Comment by gururaj1512** [comment]
@gunjjoshi, PR is ready for a review.

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# sqrtpif
+
+> Compute the principal [square root][@stdlib/math/base/special/sqrt] of the product of π and a positive single-precision floating-point number.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md:23]:
```suggestion
> Compute the principal [square root][@stdlib/math/base/special/sqrtf] of the product of π and a positive single-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+## Usage
+
+```javascript
+var sqrtpif = require( '@stdlib/math/base/special/sqrtpif' );
+```
+
+#### sqrtpif( x )
+
+Computes the principal [square root][@stdlib/math/base/special/sqrt] of the product of π and a positive single-precision floating-point number.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md:35]:
```suggestion
Computes the principal [square root][@stdlib/math/base/special/sqrtf] of the product of π and a positive single-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+// returns ~5.60499
+
+v = sqrtpif( 0.0 );
+// returns 0.0
+
+v = sqrtpif( NaN );
+// returns NaN
+```
+
+For negative numbers, the principal [square root][@stdlib/math/base/special/sqrt] is **not** defined.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md:51]:
```suggestion
For negative numbers, the principal [square root][@stdlib/math/base/special/sqrtf] is **not** defined.
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/special/sqrtpif.h"
+```
+
+#### stdlib_base_sqrtpif( x )
+
+Computes the principal [square root][@stdlib/math/base/special/sqrt] of the product of π and a positive single-precision floating-point number.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md:112]:
```suggestion
Computes the principal [square root][@stdlib/math/base/special/sqrtf] of the product of π and a positive single-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+</section>
+
+<!-- /.related -->
+
+<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="links">
+
+[@stdlib/math/base/special/sqrt]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/sqrt
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md]:
```suggestion
[@stdlib/math/base/special/sqrtf]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/sqrtf
```

**Comment by gunjjoshi** [comment]
@gururaj1512 Can you please modify the JS and C benchmarks to pre-compute the values, outside the loop? This applies to the JS examples as well.

**Comment by gunjjoshi** [comment]
Also, @gururaj1512, you'll need to add the `f` suffix after floating-point numbers in all C files in this PR.

**Comment by gururaj1512** [comment]
> Also, @gururaj1512, you'll need to add the `f` suffix after floating-point numbers in all C files in this PR.

@gunjjoshi, Updated the code as suggested. Ready for review

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/benchmark/benchmark.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var x;
+	var y;
+	var i;
+
+	x = randu( 100, 0, 100000.0 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/benchmark/benchmark.js]:
```suggestion
	x = randu( 100, 0.0, 100000.0 );
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/array/discrete-uniform' );
+var sqrtpif = require( '@stdlib/math/base/special/sqrtpif' );
+
+var x = randu( 100, 0, 100 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/README.md]:
```suggestion
var x = randu( 100, 0.0, 100.0 );
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/benchmark/benchmark.native.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+
+// MAIN //
+
+bench( pkg+'::native', opts, function benchmark( b ) {
+	var x;
+	var y;
+	var i;
+
+	x = randu( 100, 0, 100000.0 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/benchmark/benchmark.native.js]:
```suggestion
	x = randu( 100, 0.0, 100000.0 );
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/src/main.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "stdlib/math/base/special/sqrtpif.h"
+#include "stdlib/math/base/special/sqrtf.h"
+#include "stdlib/constants/float32/pi.h"
+
+/**
+* Compute the principal square root of the product of π and a positive single-precision floating-point number.
+*
+* @param x    input value
+* @return	  output value
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/examples/c/example.c [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+int main( void ) {
+	const float x[] = { 4.0f, 10.0f, 3.14f, -3.14f, 0.0f, 0.0f / 0.0f };
+
+    float v;
+    int i;
+    for ( i = 0; i < 6; i++ ) {
+        v = stdlib_base_sqrtpif( x[ i ] );
+        printf( "sqrtpif(%f) = %f", x[ i ], v );
+    }
+}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/examples/c/example.c]:
```suggestion
int main( void ) {
	const float x[] = { 4.0f, 10.0f, 3.14f, -3.14f, 0.0f, 0.0f / 0.0f };

	float v;
	int i;
	for ( i = 0; i < 6; i++ ) {
		v = stdlib_base_sqrtpif( x[ i ] );
		printf( "sqrtpif(%f) = %f", x[ i ], v );
	}
}
```

File: lib/node_modules/@stdlib/math/base/special/sqrtpif/examples/index.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+var randu = require( '@stdlib/random/array/discrete-uniform' );
+var sqrtpif = require( './../lib' );
+
+var x = randu( 100, 0, 100 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/sqrtpif/examples/index.js]:
```suggestion
var x = randu( 100, 0.0, 100.0 );
```

**Review by Planeshifter — APPROVED** [review]
Thank you @gururaj1512!

Did another review pass after @gunjjoshi's and this PR should be ready to land now.

**Pull request merged**